diff --git a/dist/tailwindcss.js b/dist/tailwindcss.js
index 10c5433b24ab91a6bbbcc87c60a6ae3d0a2a7f35..4805ca8498a84a4bb69d94274cf4bd8ddd90a272 100644
--- a/dist/tailwindcss.js
+++ b/dist/tailwindcss.js
@@ -43,7 +43,7 @@ var tailwindcss_default = (api) => {
     const inputPath = (0, import_path.join)(api.cwd, "tailwind.css");
     const generatedPath = (0, import_path.join)(api.paths.absTmpPath, outputPath);
     const binPath = getTailwindBinPath({ cwd: api.cwd });
-    const configPath = (0, import_path.join)(api.cwd, "tailwind.config.js");
+    const configPath = (0, import_path.join)(api.cwd, "tailwind.config.ts");
     if (process.env.IS_UMI_BUILD_WORKER) {
       return;
     }
