diff --git a/es/BaseInput.js b/es/BaseInput.js
index 3842dc636e53e801686934e1e3948ed8ad1dc0c6..641d4beb50db7b340aeffc051a9b931dae087fe7 100644
--- a/es/BaseInput.js
+++ b/es/BaseInput.js
@@ -52,7 +52,7 @@ var BaseInput = function BaseInput(props) {
       onMouseDown: function onMouseDown(e) {
         return e.preventDefault();
       },
-      className: clsx(clearIconCls, (_clsx = {}, _defineProperty(_clsx, "".concat(clearIconCls, "-hidden"), !needClear), _defineProperty(_clsx, "".concat(clearIconCls, "-has-suffix"), !!suffix), _clsx)),
+      className: clsx(clearIconCls, (_clsx = {}, _defineProperty(_clsx, "".concat(clearIconCls, "-has-suffix"), !!suffix), _clsx), classNames === null || classNames === void 0 ? void 0 : classNames.clear, !needClear ? 'hidden' : void 0),
       role: "button",
       tabIndex: -1
     }, iconNode);
diff --git a/es/interface.d.ts b/es/interface.d.ts
index 43ac65080a8da8f64fce366fc96d3d0b087a0d88..b5ca6507141c2767b87c03f0bee62d41a0e0956e 100644
--- a/es/interface.d.ts
+++ b/es/interface.d.ts
@@ -14,6 +14,7 @@ export interface CommonInputProps {
     classNames?: {
         prefix?: string;
         suffix?: string;
+        clear?: string;
     };
     styles?: {
         prefix?: CSSProperties;
diff --git a/lib/BaseInput.js b/lib/BaseInput.js
index ee971e68d8387bd4dd5722469539f1563b2dacae..6fc510d6f132a628dc4201eba6172236bbc926cd 100644
--- a/lib/BaseInput.js
+++ b/lib/BaseInput.js
@@ -62,7 +62,7 @@ var BaseInput = function BaseInput(props) {
       onMouseDown: function onMouseDown(e) {
         return e.preventDefault();
       },
-      className: (0, _classnames.default)(clearIconCls, (_clsx = {}, (0, _defineProperty2.default)(_clsx, "".concat(clearIconCls, "-hidden"), !needClear), (0, _defineProperty2.default)(_clsx, "".concat(clearIconCls, "-has-suffix"), !!suffix), _clsx)),
+      className: (0, _classnames.default)(clearIconCls, (_clsx = {}, (0, _defineProperty2.default)(_clsx, "".concat(clearIconCls, "-has-suffix"), !!suffix), _clsx), classNames === null || classNames === void 0 ? void 0 : classNames.clear, !needClear ? 'hidden': void 0),
       role: "button",
       tabIndex: -1
     }, iconNode);
diff --git a/lib/interface.d.ts b/lib/interface.d.ts
index 43ac65080a8da8f64fce366fc96d3d0b087a0d88..b5ca6507141c2767b87c03f0bee62d41a0e0956e 100644
--- a/lib/interface.d.ts
+++ b/lib/interface.d.ts
@@ -14,6 +14,7 @@ export interface CommonInputProps {
     classNames?: {
         prefix?: string;
         suffix?: string;
+        clear?: string;
     };
     styles?: {
         prefix?: CSSProperties;